extension/dhcpdump/dhcpdump.js

```diff
@@ -103,28 +103,35 @@ OPTION:  12 ( 12) Host name                 Great-Room-3
   parse(output) {
      let o =  output.split(/\r?\n/);
      let obj = {};
+     var foundMac = null
     for (let i in o) {
       let line = o[i];
 
       // locate mac address
       // from "IP: 0.0.0.0 (2:42:ac:11:0:2) > 255.255.255.255 (ff:ff:ff:ff:ff:ff)"
       let match = line.match("IP: .* \\((.*)\\) > 255.255.255.255");
       if(match) {
-        obj.mac = this.normalizeMac(match[1]);
+        foundMac = this.normalizeMac(match[1])
       }
 
       // locate hostname
       let match2 = line.match("OPTION:.{1,9}12.{1,9}Host name +([^ ]+)");
       if(match2) {
-        obj.name = match2[1];
+        if (foundMac != null) {
+          obj[foundMac] = match2[1];
+        }
       }
-    }
 
-    if(obj.mac && obj.name) {
-      return obj;
-    } else {
-      return {};
+      //for multiple events,
+      //when match the split line '---------------------------' between two adjacent events,
+      //need reset the foundMac in case one dhcp event has mac without hostname,
+      //then the next event has hostname without mac
+      if (line.indexOf("------------------------------------------------") >= 0) {
+        foundMac = null
+      }
     }
+
+    return obj
```

Comment:
general comment: this implementation is okay, and a better design is spliting data to multiple sub-datas (each sub data contains a single dhcp event), and have this.parse(subdata) to extract the mac and hostname. This way the parse function will specifically focus on parsing single dhcp event, which makes the logic simple.

Reply:
```javascript
let msgs = message.split("------------------------------------------------")

msgs.forEach((m) => {
  let obj = this.parse(m)
  callback(obj)
})
```